lib/logstash/outputs/sns.rb

```diff
@@ -70,54 +63,45 @@ def register
   def receive(event)
     return unless output?(event)
 
-    arn     = Array(event["sns"]).first || @arn
-
-    raise "An SNS ARN required." unless arn
-
-    message = Array(event["sns_message"]).first
-    subject = Array(event["sns_subject"]).first || event["host"]
-
-    # Ensure message doesn't exceed the maximum size.
-    if message
-      # TODO: Utilize `byteslice` in JRuby 1.7: http://jira.codehaus.org/browse/JRUBY-5547
-      message = message.slice(0, MAX_MESSAGE_SIZE_IN_BYTES)
+    if (sns_msg = Array(event["sns_message"]).first)
+      send_sns_message(event_arn(event), event_subject(event), sns_msg)
```

Comment:
Any fields can be an `Array` or a `Hash`, I think what we should do here is to allow the user to set as an option which fields they want for the `sns_message` or `sns_subject` and we assume that the field is a string. 

By doing `.first` we already do a decision for the user, if they want to use the first value of the array they can still do it:

```ruby
irb(main):006:0> require "logstash/event"
=> false
irb(main):007:0> e = LogStash::Event.new({"awesome" => ["one", "two"], "notawesome" => "wifi-on-train"})
=> #<LogStash::Event:0x4aee6e0a @metadata_accessors=#<LogStash::Util::Accessors:0x36e0a11d @store={}, @lut={}>, @cancelled=false, @data={"awesome"=>["one", "two"], "notawesome"=>"wifi-on-train", "@version"=>"1", "@timestamp"=>"2015-06-22T18:30:28.766Z"}, @metadata={}, @accessors=#<LogStash::Util::Accessors:0x2ed46cd5 @store={"awesome"=>["one", "two"], "notawesome"=>"wifi-on-train", "@version"=>"1", "@timestamp"=>"2015-06-22T18:30:28.766Z"}, @lut={}>>
irb(main):008:0> e["[awesome][0]"]
=> "one"
irb(main):009:0> e["[notawesome][0]"]
=> nil
irb(main):010:0>
```

This change isn't backward compatible, but I think this is an uncommon situation..
Also since we uses v2 api and we now uses codec which are major changes I believe the plugin should be released as a new major version.

Maybe other people can comment on this?